mypy/test/teststubtest.py

```diff
@@ -757,7 +761,14 @@ def h(x: str): ...
             runtime="",
             error="h",
         )
-        yield Case(stub="", runtime="__all__ = []", error=None)  # dummy case
+        # If runtime has __all__ but stub does not,
+        # we'll already raise an error for a missing __all__,
+        # so we shouldn't raise another error for inconsistent exported names
+        yield Case(stub="_Z = int", runtime="__all__ = []", error=None)
```

Comment:
Unfortunately, test cases within a `collect_cases` aren't that independent, since they end up in the same file. For all of these, the stub has an `__all__`. Big gotcha about how stubtest does tests, but it does make them much faster :-)

Reply:
Argh, of course!